.github/workflows/build-images.yaml

```diff
@@ -9,523 +9,540 @@ on:
       - master
       - dev
 
+env:
+  PUSH_COMMIT_RANGE: ${{ github.event.before }}...${{ github.event.after }}
+  PR_COMMIT_RANGE: ${{ github.event.pull_request.base.sha }}...${{ github.event.pull_request.head.sha }}
+  DOCKER_USER: ${{ secrets.DOCKER_HUB_USERNAME }}
+  DOCKER_PASS: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+  FORCE_PUSH: "false"
+
 jobs:
   base-os:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        target-os: [amd64, arm64]
+    env:
+      DH_REPO: "cloudsuitetest/debian"
     steps:
       - name: checkout
         uses: actions/checkout@v2
-        
-      - name: Login to Docker Hub
-        uses: docker/login-action@v1
         with:
-          username: ${{ secrets.DOCKER_HUB_USERNAME }}
-          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
-          
+          fetch-depth: 0
       - name: setup docker buildx
         uses: docker/setup-buildx-action@v1
       - name: build and push
-        uses: docker/build-push-action@v2
-        with:
-          context: ./commons/${{ github.job }}
-          platforms: linux/${{ matrix.target-os }}
-          file: ./commons/${{ github.job }}/Dockerfile.${{ matrix.target-os }}
-          push: true
-          tags: "cloudsuitetest/debian:${{ matrix.target-os }}"
-      - run: docker manifest create --amend cloudsuitetest/debian:base-os cloudsuitetest/debian:amd64 cloudsuitetest/debian:arm64
-      - run: docker manifest push cloudsuitetest/debian:base-os
+        run: "./.github/scripts/build-images.sh"
   
   java:
     runs-on: ubuntu-latest
     needs: base-os
+    env:
+      DH_REPO: "cloudsuitetest/${{ github.job }}"
     strategy:
       matrix:
-        tags: [openjdk11, adoptopenjdk8]
+        include:
+          - tag: "adoptopenjdk8"
+            platform: "linux/amd64,linux/arm64"
+          - tag: "openjdk11"
+            platform: "linux/amd64,linux/arm64,linux/riscv64"
     steps:
       - name: checkout
         uses: actions/checkout@v2
-      - name: Login to Docker Hub
-        uses: docker/login-action@v1
         with:
-          username: ${{ secrets.DOCKER_HUB_USERNAME }}
-          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+          fetch-depth: 0
       - name: setup docker buildx
```

Comment:
I have a question regarding this.
Do we need this extra step? I mean can't we do whatever this has to do inside our script?
Then we don't need to repeat writing the step n times for each of the images.

Reply:
I actually prefer keeping this step as if it's referring to some outside library maintained by others. It's definitely possible to do it by ourselves but we have to maintain it.

Reply:
OK, let's leave this as is for now and we will get back to it later if needed.